### PR TITLE
Refactors event handler

### PR DESCRIPTION
**Pull request by SuchAFuriousDeath**
I guess VmmEventHandle will have to wrap an impl FnMut(VmmEvent) + Clone closure? Or something like that.

### CODE REVIEW

**Comment by ultimaweapon** [comment]
> I guess VmmEventHandle will have to wrap an impl FnMut(VmmEvent) + Clone closure?

Should be `Arc<dyn Fn(...)>` instead.

**Comment by SuchAFuriousDeath** [comment]
> > I guess VmmEventHandle will have to wrap an impl FnMut(VmmEvent) + Clone closure?
> 
> Should be `Arc<dyn Fn(...)>` instead.

Close enough :sweat_smile:

**Comment by ultimaweapon** [comment]
I don't think global models are a good idea. Why do we need it? It should be passing to the main window and downstream to all of its children instead.

**Comment by SuchAFuriousDeath** [comment]
> > I don't think global models are a good idea. Why do we need it? It should be passing to the main window and downstream to all of its children instead.
> 
> When launching in debug, there is no main window (and the screen isn't its child anyway). Maybe that's the fix that needs fixing here 🤔

But that's actually not the real problem. The real problem is that I don't like having to pass the devices and profiles through several elements down to where we actually need them. If they're global, we can just directly access them right where we need them (in the ComboBoxes)

**Comment by ultimaweapon** [comment]
> > > I don't think global models are a good idea. Why do we need it? It should be passing to the main window and downstream to all of its children instead.
> > 
> > 
> > When launching in debug, there is no main window (and the screen isn't its child anyway). Maybe that's the fix that needs fixing here 🤔
> 
> But that's actually not the real problem. The real 'problem' is that I don't like having to pass the devices and profiles through several elements down to where we actually need them. If they're global, we can just directly access them right where we need them (in the ComboBoxes)

What you don't like here are actually a good practice.

**Comment by SuchAFuriousDeath** [comment]
> > > > I don't think global models are a good idea. Why do we need it? It should be passing to the main window and downstream to all of its children instead.
> > > 
> > > 
> > > When launching in debug, there is no main window (and the screen isn't its child anyway). Maybe that's the fix that needs fixing here 🤔
> > 
> > 
> > But that's actually not the real problem. The real 'problem' is that I don't like having to pass the devices and profiles through several elements down to where we actually need them. If they're global, we can just directly access them right where we need them (in the ComboBoxes)
> 
> What you don't like here are actually a good practice.

Oops :D

File: gui/src/main.rs [hunk 1]
```diff
@@ -78,24 +89,28 @@ fn run() -> Result<(), ApplicationError> {
...
-    setup_global_callbacks(&main_window);
-
     let graphics_api = graphics::DefaultApi::new().map_err(ApplicationError::InitGraphicsApi)?;
 
-    let devices: Vec<SharedString> = graphics_api
-        .physical_devices()
-        .into_iter()
-        .map(|d| SharedString::from(d.name()))
-        .collect();
+    let profiles = vec![Profile::default()];
```
Comment (ultimaweapon) [inline gui/src/main.rs:94]: Better to centralize this now so we don't have a duplicated code.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**